### DISCUSSION TITLE
OVN-Kubernetes v1.0.0

### DISCUSSION THREAD

**Discussion by tssurya** [ovn-org/ovn-kubernetes#4444] [Announcements]
# v1.0.0

📢 We are happy to announce the release of OVN-Kubernetes v1.0.0, our first major release.  🎉 🎊 🎆 🎇💥
This version of OVN-Kubernetes uses:
☸️ Kubernetes v1.29.2
🔁 OVN ovn-24.03.2-5.fc39.x86_64
🔄 OVS openvswitch-3.2.2-1.fc39.x86_64

## Summary of important changes for this release

This release includes many exciting features that end-users can leverage:

🔀 **Interconnect** mode of deploying OVN-Kubernetes with a distributed database architecture - this makes OVN-Kubernetes more scalable, performant and secure

🌐 **EgressIPs** which allows users to send northbound traffic from the selected pods using "the special pool of user configured egressIPs"

🔐 **EgressFirewall** which allows users to impose fine grained traffic security rules on northbound traffic leaving from the selected pods in the cluster

🎯 **EgressQoS** which performs DSCP markings on pods that can be leveraged to perform traffic shaping and policing

🎖️ **EgressService** which allows users to send northbound traffic from the selected pods using the load balancer service ingress VIP (reverse of service ingress)

🌐 **Multiple External Gateways** which allows pods to send egress traffic outside the cluster via a set of defined egress gateway pods

🔐 **AdminNetworkPolicy** which allows admins to create cluster-wide security policies that limit network traffic based on the principle of least privilege and zero trust

💫  **Multihoming** which allows plugging multiple interfaces to a Pod with each of the secondary interfaces managed by OVN-Kubernetes CNI

🔐 **MultiNetworkPolicy** provides an ability to define network policies on the secondary interfaces of the pods

📥 **HardwareAcceleration** provides an ability to achieve line speed Kubernetes Pod networking - transmit/receive, network policy, NATing, Load Balancers, and so on

🏗️ **Kubevirt VM Live Migration** allows [kubevirt](https://kubevirt.io/) virtual machines to be [live migrated](https://kubevirt.io/user-guide/operations/live_migration/) while keeping the established TCP connections alive, and preserving the VM IP configuration.

🌉 **Hybrid Overlay** provides the ability to run both linux and windows nodes on your same cluster

🚇 **Tunneling over Multiple NICS** provides an ability to tunnel traffic from K8s Pods over different NICs on the same host.

Check out the [ovn-kubernetes website](https://ovn-kubernetes.io/) for more information. ☸️

Please reach out to us and provide feedback based on your experience of using this release so that we can keep improving this project!

Full Changelog: https://github.com/ovn-org/ovn-kubernetes/compare/v0.3.0...release-1.0

## Contributors 💖 

Many thanks to all of our contributors who helped make this happen 😄👏 !! We truly couldn't have done this without all your contributions. Contributors info taken based on [this data](https://github.com/ovn-org/ovn-kubernetes/graphs/contributors?from=2018-05-11&to=2024-06-14&type=c).

@girishmg @trozet @dcbw @tssurya @alexanderConstantinescu @npinaeva @jcaamano @martinkennelly @JacobTanenbaum @maiqueb @danwinship @andreaskaris @kyrtapz @flavio-fernandes @astoycos @cathy-zhou @ricky-rav @oribon @fedepaol @shettyg @squeed @dave-tucker @pecameron @ionutbalutoiu @numansiddique @adrianchiris @pperiyasamy @dceara @abhat @aserdean @jordigilh @Pardhakeswar @qinqon @bpickard22 @Billy99 @alinbalutoiu @jluhrsen @vjayaramrh @russellb @pliurh @zshi-redhat @nerdalert @zeeke @as-com @hzhou8 @creydr @roytman @msherif1234 @winsopc @vishnoianil @oshoval @moshe010 @wizhaoredhat @gvbalaji @igsilya @bn222 @shahar-klein @arghosh93 @lanoxx @DmytroLinkin @mccv1r0 @TrevorTaoARM @rcarrillocruz @arkadeepsen @markmc @skitt @vpickard @jotak @masap @dougsland @anfredette @aojea @lmilleri @josecastillolema @sdsen @ganboing @rsevilla87 @stbenjam @amorenoz @haojue @rbbratta @Reamer @krescent @danielmellado @mansikulkarni96 @venuiyer @JocelynBerrendonner @SchSeba @JM1 @hareeshpc @csahithi @dtzhou2 

## Docker Image Manifests

Our images can be downloaded from the GH Container registry [here](https://github.com/ovn-org/ovn-kubernetes/pkgs/container/ovn-kubernetes%2Fovn-kube-fedora/230290476?tag=release-1.0).

## Binaries Released

- `ovnkube` which is our main single all-in-one binary executable used to launch the ovnkube control plane and data plane pods in a kubernetes deployment
- `ovn-k8s-cni-overlay` is the cni executable to be placed in /opt/cni/bin (or another directory in which kubernetes will look for the plugin) so that it can be invoked for each pod event by kubernetes 
- `hybrid-overlay-node` is a node controller to integrate disparate networks with VXLAN tunnels
- `ovn-kube-util` contains the Utils for ovn-kubernetes
- `ovndbchecker` runs ovn db checker to ensure raft membership and db health on default deployment modes
- `ovnkube-trace` is the binary that contains ovnkube-trace which is an abstraction used to invoke OVN/OVS packet tracing utils
- `ovnkube-identity` is the executable that is invoked to run ovn-kubernetes identity manager, which includes the admission webhook and the CertificateSigningRequest approver

<hr /><em>This discussion was created from the release <a href='https://github.com/ovn-org/ovn-kubernetes/releases/tag/v1.0.0'>OVN-Kubernetes v1.0.0</a>.</em>

**Comment by garca-cloud** [comment]
Hi all, 
I'm following this project and I'd like to have more information about:

- Installation on k8s test env; there are many ways to install, but all dosen't work on my cluster.
- Documentation on website: I found many empty page
- Multi tenancy funtions (like VPC): I want to test the client isolation and the network overlapping.

It's possibile to have some clarifications by an email or slack channel?
thanks.

**Reply by tssurya** [reply]
@garca-cloud welcome! Our website is relatively new - we are still trying to bridge our gaps, but we have bi-weekly meetings: https://ovn-kubernetes.io/governance/MEETINGS/ where everyone is welcome to join

We also just joined CNCF hence we are about to get a new slack workspace/channel but for now please feel free to tell me your email and I can invite you to our slack channel.

For Multi-tenancy we have many features for VPC like - perhaps user defined network segmentation is what you are after -> you can split your cluster into multiple isolated networks, these networks can have overlapping subnets.

**Reply by garca-cloud** [reply]
Hi @tssurya, thanks to your reply.
My email is giuseppe.arcarese@telecomitalia.it and it's a pleasure to join in the dedicated slack channel.

**Comment by tssurya** [comment]
@garca-cloud : you have been invited! please feel free to start a thread in ovn-kubernetes channel and we can help you
